### ISSUE TITLE
vrrp_script can't handle error codes different than 0 or 1 (since 1.2.19 -> 1.2.24 upgrade)

### ISSUE THREAD

**Issue by chr4** [acassen/keepalived#1027]
I'm sorry that this concerns an older version (1.2.24 running on Ubuntu LTS 16.04) - I'm not sure if this is already addressed. I didn't find anything when searching Google/ this repos issues, though.

After the upgrade (via `apt-get dist-upgrade`) from 1.2.19 to 1.2.24, I've experienced weird behaviour of my existing keepalived configuration. Among others, these were the relevant lines from the logs:

```
Keepalived[xxxx1]: VRRP child process(xxxx5) died: Respawning
Keepalived_vrrp[xxxx2]: pid xxxx3 exited with status 256
Keepalived_vrrp[xxxx2]: pid xxxx4 exited with permanent error FATAL. Terminating
```

After a bit of digging, I think I've found the error: When defining a `vrrp_script`, keepalived crashes (?) when script returns an error message different to `0` or `1`. Here's my configuration:

```
vrrp_script script_name {
    script "/path/to/script"
    interval 2
    weight 100
}
```

The relevant section of the script `script_name`:

```bash
systemctl status myservice > /dev/null
exit $?
```

When catching `$?` and returning `0` and `1` instead, the crash doesn't happen:

```bash
systemctl status myservice > /dev/null
if [ $? != 0 ]; then exit 1; else exit 0; fi
```

This looks like an error to me. If this is already addressed (since the version supported by Ubuntu is quite old), I hope this at least helps people experiencing the same issues.

**Comment by pqarmitage** [comment]
Commit c0f8c1a resolved this issue. This commit was first included in v1.4.0.

The problem only occurs with exit codes 2 and 3, and was caused by keepalived not distinguishing between the exit status of scripts and the exit status detected by the parent process of the vrrp and checker child processes.

**Comment by chr4** [comment]
Thanks for the quick response and explanation!

I was wondering, there's no official keepalived repository for Debian/ Ubuntu, right?
The current version of Ubuntu LTS (18.04) is still on v1.3.9 :(
